### ISSUE TITLE
Warning about “writing to an object of type 'struct NoiseParams' with no trivial copy-assignment” in file src/noise.cpp when compiling

### ISSUE THREAD

**Issue by ghost** [minetest/minetest#8292]
##### Issue type

 - Build issue
 
##### Minetest version

```
$ git rev-parse HEAD
111f1dc9c5f336f5b4f12c15940dc4cc2cbe7f82
```

##### OS / Hardware

Operating system: Linux 4.20.6-arch1-1-ARCH x86_64
CPU: Intel(R) Core(TM) i7-4790K CPU @ 4.00GHz

GNU Make 4.2.1
cmake version 3.13.4

##### Summary
When compiling Minetest as described in the readme at one point the following warning appears.

```
/home/dirk/test/minetest/src/noise.cpp: In constructor 'Noise::Noise(NoiseParams*, irr::s32, irr::u32, irr::u32, irr::u32)':
/home/dirk/test/minetest/src/noise.cpp:432:29: warning: 'void* memcpy(void*, const void*, size_t)' writing to an object of type 'struct NoiseParams' with no trivial copy-assignment; use copy-assignment or copy-initialization instead [-Wclass-memaccess]
  memcpy(&np, np_, sizeof(np));
                             ^
In file included from /home/dirk/test/minetest/src/noise.cpp:27:
/home/dirk/test/minetest/src/noise.h:103:8: note: 'struct NoiseParams' declared here
 struct NoiseParams {
        ^~~~~~~~~~~
```

##### Steps to reproduce
1. Compile the latest available version (when writing this it’s 111f1dc9c5f336f5b4f12c15940dc4cc2cbe7f82)
2. Watch the build output

**Comment by paramat** [comment]
This happened to me a while back in a PR, investigating.

**Comment by paramat** [comment]
Same warning as #8117 
Fix is https://github.com/minetest/minetest/issues/8117#issuecomment-456175520

**Comment by paramat** [comment]
I'm concerned about making a change close to release though.

**Comment by Linuxdirk** [comment]
Since this is not a blocker I think having this fixed in a bugfix release after 5.0.0 is absolutely fine. Same with https://github.com/minetest/minetest/issues/8291 (I just noticed them while watching the build process).

**Comment by juozaspo** [comment]
I also noticed this issue happening in recent Ubuntu development version (20.04, future LTS release) with gcc 9.2.1, cmake 3.16.3 and make 4.2.1
Affected commit: 8ddb6718
